rust/core-lib/src/backspace.rs

```diff
@@ -16,23 +16,17 @@
 use xi_rope::{Cursor, Rope};
 
 use crate::config::BufferItems;
-use crate::line_offset::LineOffset;
+use crate::line_offset::{DefaultLineOffset, LineOffset};
 use crate::selection::SelRegion;
-use crate::view::View;
 use xi_unicode::*;
 
 #[allow(clippy::cognitive_complexity)]
-pub fn offset_for_delete_backwards(
-    view: &View,
-    region: &SelRegion,
-    text: &Rope,
-    config: &BufferItems,
-) -> usize {
+pub fn offset_for_delete_backwards(region: &SelRegion, text: &Rope, config: &BufferItems) -> usize {
     if !region.is_caret() {
         region.min()
     } else {
         // backspace deletes max(1, tab_size) contiguous spaces
-        let (_, c) = view.offset_to_line_col(&text, region.start);
+        let (_, c) = DefaultLineOffset.offset_to_line_col(&text, region.start);
```

Comment:
this seems to be changing the behaviour? by not using the view we may get different results here when word wrap is enabled. I'm not sure if this is a problem in practice, but it feels fishy.

Reply:
`view.line_of_offset` calls `lines.visual_line_of_offset`, so different behaviour on word wrap was the default and the change a bugfix (see PR summary).
I've renamed `DefaultLineOffset` to `LogicalLines`, which should make the change more obvious (and `DefaultLineOffset` more descriptive).